src/main/java/org/prebid/server/bidder/cpmstar/CPMStarBidder.java

```diff
@@ -0,0 +1,164 @@
+            }
+        }
+        return bidRequest.toBuilder().imp(validImpList).build();
+    }
+
+    private Imp parseAndValidateImp(Imp imp) {
+        if (imp.getBanner() == null && imp.getVideo() == null) {
+            throw new PreBidException("Only Banner and Video bid-types are supported at this time");
+        }
+
+        final ExtPrebid<?, ExtImpCPMStar> ext =
+                Json.mapper.convertValue(imp.getExt(), CPM_STAR_EXT_TYPE_REFERENCE);
+
+        final ExtImpCPMStar bidder = ext.getBidder();
+        if (bidder == null) {
+            throw new PreBidException(String.format("imp id=%s: bidder.ext is null", imp.getId()));
+        }
+        return imp.toBuilder().ext(Json.mapper.valueToTree(ext)).build();
+    }
+
+    private HttpRequest<BidRequest> createSingleRequest(BidRequest request) {
+
+        final String body = Json.encode(request);
+
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(HttpUtil.headers())
+                .body(body)
+                .payload(request)
+                .build();
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        if (httpCall.getResponse().getStatusCode() == HttpResponseStatus.NO_CONTENT.code()) {
+            return Result.of(Collections.emptyList(), Collections.emptyList());
+        }
+
+        try {
+            final BidResponse bidResponse = Json.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return extractBids(httpCall.getRequest().getPayload(), bidResponse);
+        } catch (DecodeException e) {
+            return Result.emptyWithError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private Result<List<BidderBid>> extractBids(BidRequest request, BidResponse bidResponse) {
+        if (bidResponse == null || bidResponse.getSeatbid() == null) {
+            return Result.of(Collections.emptyList(), Collections.emptyList());
+        }
+        final List<BidderError> errors = new ArrayList<>();
+        final List<BidderBid> bidderBids = bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> bidFromResponse(request.getImp(), bid, errors))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        return Result.of(bidderBids, errors);
+    }
+
+    private static BidderBid bidFromResponse(List<Imp> imps, Bid bid, List<BidderError> errors) {
+        try {
+            final BidType bidType = getBidType(bid.getImpid(), imps);
+            return BidderBid.of(bid, bidType, DEFAULT_BID_CURRENCY);
+        } catch (PreBidException e) {
+            errors.add(BidderError.badInput(e.getMessage()));
+            return null;
```

Comment:
I know that we have similar logic in another bidders, but I would rewrite something like that
```
    private Result<List<BidderBid>> extractBids(BidRequest request, BidResponse bidResponse) {
        if (bidResponse == null || bidResponse.getSeatbid() == null) {
            return Result.of(Collections.emptyList(), Collections.emptyList());
        }
        final List<Bid> responseBids = bidResponse.getSeatbid().stream()
                .filter(Objects::nonNull)
                .map(SeatBid::getBid)
                .filter(Objects::nonNull)
                .flatMap(Collection::stream)
                .collect(Collectors.toList());

        final List<BidderError> errors = new ArrayList<>();
        return Result.of(bidsFromResponse(request.getImp(), responseBids, errors), errors);
    }

    private static List<BidderBid> bidsFromResponse(List<Imp> imps, List<Bid> responseBids, List<BidderError> errors) {
        final List<BidderBid> bidderBids = new ArrayList<>();
        for (Bid responseBid : responseBids) {
            try {
                final BidType bidType = resolveBidType(responseBid.getImpid(), imps);
                bidderBids.add(BidderBid.of(responseBid, bidType, DEFAULT_BID_CURRENCY));
            } catch (PreBidException e) {
                errors.add(BidderError.badInput(e.getMessage()));
            }
        }
        return bidderBids;
    }
```

Just to get rid of `return null` (looks to me like GO :hankey:)